cognite/client/data_classes/datapoints.py

```diff
@@ -189,15 +208,74 @@ def to_pandas(self, column_names="externalId") -> "pandas.DataFrame":
             identifier = self.id
         else:
             raise ValueError("column_names must be 'externalId' or 'id'")
-        for attr, value in self._get_non_empty_data_fields(get_empty_lists=True):
+        fields = self._get_non_empty_data_fields(get_empty_lists=True)
+        for attr, value in fields:
             if attr == "timestamp":
                 timestamps = value
             else:
                 id_with_agg = str(identifier)
                 if attr != "value":
-                    id_with_agg += "|{}".format(utils._auxiliary.to_camel_case(attr))
+                    id_with_agg = "{}|{}".format(id_with_agg, utils._auxiliary.to_camel_case(attr))
                 data_fields[id_with_agg] = value
-        return pd.DataFrame(data_fields, index=pd.DatetimeIndex(data=np.array(timestamps, dtype="datetime64[ms]")))
+
+        df = pd.DataFrame(data_fields, index=pd.DatetimeIndex(data=np.array(timestamps, dtype="datetime64[ms]")))
+        df = self._complete_dataframe_column(
+            df, complete, complete_start, complete_end, complete_is_step, [attr for attr, value in fields]
+        )
+
+        if not include_aggregate_name:
+            if len(fields) > 2:  # timestamp + 1
+                raise ValueError(
+                    "include_aggregate_name=False can not be used here as it would lead to duplicate column names for multiple aggregates {}".format(
+                        fields
+                    )
+                )
+            df.rename(columns=lambda s: regexp.sub(r"\|\w+$", "", s), inplace=True)
+        return df
+
+    def _complete_dataframe_column(self, df, complete, complete_start, complete_end, complete_is_step, aggregates):
+        np, pd = utils._auxiliary.local_import("numpy", "pandas")
+        complete = Datapoints._get_complete_options(complete)
+        if "fill" in complete and df.shape[0] > 1:
+
+            if complete_is_step is None and "interpolation" in aggregates:
+                client = getattr(self, "_cognite_client", None)
+                if client:
+                    complete_is_step = Datapoints._is_step(client, [self.id])
+            complete_granularity = getattr(self, "_granularity", None)
+            if not complete_granularity:
+                raise AttributeError("Can only complete Datapoints objects returned from DatapointsAPI.retrieve")
```

Comment:
Raising an AttributeError here is a little off. I would create a custom exception for this.